caluma/caluma_form/tests/test_answer.py

```diff
@@ -312,3 +313,38 @@ def validate(self, mutation, data, info):
         result.data["saveDocumentStringAnswer"]["answer"]["stringValue"]
         == "Test (validated)"
     )
+
+
+@pytest.mark.parametrize("question__type", [Question.TYPE_CALCULATED_FLOAT])
+@pytest.mark.parametrize("use_graphql", [True, False])
+def test_validate_save_calculated_float_answer(
+    db, mocker, document, question, schema_executor, use_graphql
+):
+    query = """
+        mutation saveDocumentCalculatedFloatAnswer($input: SaveDocumentCalculatedFloatAnswerInput!) {
+          saveDocumentStringAnswer(input: $input) {
+            answer {
+              __typename
+              ... on CalculatedFloatAnswer {
+                value
+              }
+            }
+            clientMutationId
+          }
+        }
+    """
+
+    variables = {
+        "input": {
+            "document": to_global_id("CalculatedFloatAnswer", document.pk),
+            "question": to_global_id("CalculatedFloatAnswer", question.pk),
+            "value": 1.0,
+        }
+    }
+
+    if use_graphql:
+        result = schema_executor(query, variable_values=variables)
+        assert result.errors
```

Comment:
I don't think you're testing what you think you're testing ;-)

The type `CalculatedFloatAnswer` doesn't exist, so you're probably testing schema validation here instead of checking your actual code.

I suggest asserting the exact content of the errors here, or,  seeing that the schema doesn't even have the required types save a calculated answer, you may also just ditch the graphql part in this test

Reply:
Done